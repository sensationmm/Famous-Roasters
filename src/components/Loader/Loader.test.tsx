import { render, waitFor } from '@testing-library/react'
import React from 'react'

import { Loader } from '.'

describe('Loader component', () => {
  it('Renders correctly', async () => {
    const { container } = render(<Loader />)
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly in small', async () => {
    const { container } = render(<Loader isSmall />)
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })

  it('Renders correctly with center false', async () => {
    const { container } = render(<Loader center={false} />)
    await waitFor(() => new Promise((res) => setTimeout(res, 0)))
    expect(container).toMatchSnapshot()
  })
})
