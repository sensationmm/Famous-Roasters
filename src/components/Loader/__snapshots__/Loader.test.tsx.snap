// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader component Renders correctly 1`] = `
<div>
  <div
    class="flex justify-center"
  >
    <svg
      class="text-brand-grey-mine animate-spin fill-brand-grey-bombay w-12 h-12"
      fill="none"
      role="status"
      viewBox="0 0 58 58"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M54 29C54 34.2537 52.3449 39.3739 49.2697 43.6335C46.1944 47.8932 41.8554 51.0758 36.8687 52.7294C31.882 54.383 26.5011 54.4236 21.4901 52.8454C16.479 51.2672 12.0925 48.1504 8.95337 43.9376C5.81425 39.7249 4.08209 34.6302 4.00284 29.3771C3.9236 24.124 5.5013 18.9794 8.51192 14.6739C11.5225 10.3683 15.8131 7.12065 20.7742 5.39202C25.7354 3.66339 31.1151 3.54164 36.1494 5.04408"
        fill="none"
        stroke="black"
        stroke-linecap="round"
        stroke-width="7"
      />
    </svg>
  </div>
</div>
`;

exports[`Loader component Renders correctly in small 1`] = `
<div>
  <div
    class="flex justify-center"
  >
    <svg
      class="text-brand-grey-mine animate-spin fill-brand-grey-bombay w-6 h-6"
      fill="none"
      role="status"
      viewBox="0 0 58 58"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M54 29C54 34.2537 52.3449 39.3739 49.2697 43.6335C46.1944 47.8932 41.8554 51.0758 36.8687 52.7294C31.882 54.383 26.5011 54.4236 21.4901 52.8454C16.479 51.2672 12.0925 48.1504 8.95337 43.9376C5.81425 39.7249 4.08209 34.6302 4.00284 29.3771C3.9236 24.124 5.5013 18.9794 8.51192 14.6739C11.5225 10.3683 15.8131 7.12065 20.7742 5.39202C25.7354 3.66339 31.1151 3.54164 36.1494 5.04408"
        fill="none"
        stroke="black"
        stroke-linecap="round"
        stroke-width="7"
      />
    </svg>
  </div>
</div>
`;

exports[`Loader component Renders correctly with center false 1`] = `
<div>
  <div>
    <svg
      class="text-brand-grey-mine animate-spin fill-brand-grey-bombay w-12 h-12"
      fill="none"
      role="status"
      viewBox="0 0 58 58"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M54 29C54 34.2537 52.3449 39.3739 49.2697 43.6335C46.1944 47.8932 41.8554 51.0758 36.8687 52.7294C31.882 54.383 26.5011 54.4236 21.4901 52.8454C16.479 51.2672 12.0925 48.1504 8.95337 43.9376C5.81425 39.7249 4.08209 34.6302 4.00284 29.3771C3.9236 24.124 5.5013 18.9794 8.51192 14.6739C11.5225 10.3683 15.8131 7.12065 20.7742 5.39202C25.7354 3.66339 31.1151 3.54164 36.1494 5.04408"
        fill="none"
        stroke="black"
        stroke-linecap="round"
        stroke-width="7"
      />
    </svg>
  </div>
</div>
`;
